language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/inlayhint/InlayHintProvider.java

```diff
@@ -128,12 +146,68 @@ private static Pair<List<ParameterSymbol>, Optional<ParameterSymbol>> getParamet
             // Since the method is a lang-lib function, skip the first parameter
             return Pair.of(libFunction.get().params().get().stream().skip(1).collect(Collectors.toList()),
                     libFunction.get().restParam());
+        } else if (invokableNode.kind() == SyntaxKind.CLIENT_RESOURCE_ACCESS_ACTION) {
+            ClientResourceAccessActionNode resourceAccessActionNode = (ClientResourceAccessActionNode) invokableNode;
+            Optional<Symbol> symbol = context.currentSemanticModel().get().symbol(resourceAccessActionNode);
+            FunctionTypeSymbol typeSymbol = ((ResourceMethodSymbol) symbol.get()).typeDescriptor();
+            return symbol.map(methodSymbol -> Pair.of(typeSymbol.params().orElse(Collections.emptyList()),
+                    typeSymbol.restParam())).orElse(Pair.of(Collections.emptyList(), Optional.empty()));
+        } else if (invokableNode.kind() == SyntaxKind.REMOTE_METHOD_CALL_ACTION) {
+            RemoteMethodCallActionNode methodCallActionNode = (RemoteMethodCallActionNode) invokableNode;
+            Optional<Symbol> symbol = context.currentSemanticModel().get().symbol(methodCallActionNode);
+            FunctionTypeSymbol typeSymbol = ((MethodSymbol) symbol.get()).typeDescriptor();
+            return symbol.map(methodSymbol -> Pair.of(typeSymbol.params().orElse(Collections.emptyList()),
+                    typeSymbol.restParam())).orElse(Pair.of(Collections.emptyList(), Optional.empty()));
+        } else if (invokableNode.kind() == SyntaxKind.IMPLICIT_NEW_EXPRESSION) {
+            return getParameterSymbolsForImplicitExpressions(context, invokableNode);
+        } else if (invokableNode.kind() == SyntaxKind.EXPLICIT_NEW_EXPRESSION) {
+            return getParameterSymbolsForExplicitExpressions(context, invokableNode);
         } else {
             FunctionCallExpressionNode functionCallExpressionNode = (FunctionCallExpressionNode) invokableNode;
             return getParameterSymbolsForFunctionCall(context, functionCallExpressionNode);
         }
     }
 
+    private static Pair<List<ParameterSymbol>, Optional<ParameterSymbol>> getParameterSymbolsForImplicitExpressions(
+            InlayHintContext context,
+            NonTerminalNode node) {
+        ImplicitNewExpressionNode implicitNode = (ImplicitNewExpressionNode) node;
```

Comment:
Why do we need to do a cast here?

Reply:
Fixed in [69bb795](https://github.com/ballerina-platform/ballerina-lang/pull/40868/commits/69bb7953c7d3955b20525ed35b6e1d672eaeb07d)